kotlin/viewshed-geoprocessing/src/main/java/com/esri/arcgisruntime/sample/viewshedgeoprocessing/MainActivity.kt

```diff
@@ -0,0 +1,216 @@
+import com.esri.arcgisruntime.symbology.SimpleMarkerSymbol
+import com.esri.arcgisruntime.symbology.SimpleRenderer
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingFeatures
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingJob
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingParameters
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingTask
+import kotlinx.android.synthetic.main.activity_main.*
+import java.util.concurrent.ExecutionException
+import kotlin.math.roundToInt
+
+class MainActivity : AppCompatActivity() {
+
+  private val geoprocessingTask: GeoprocessingTask by lazy { GeoprocessingTask(getString(R.string.viewshed_service)) }
+  private var geoprocessingJob: GeoprocessingJob? = null
+
+  private val inputGraphicsOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+  private val resultGraphicsOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create a map with the Basemap type topographic
+    val map = ArcGISMap(Basemap.Type.TOPOGRAPHIC, 45.3790902612337, 6.84905317262762, 12)
+    // set the map to be displayed in this view
+    mapView.map = map
+
+    // renderer for graphics overlays
+    val fillColor = Color.argb(120, 226, 119, 40)
+    val fillSymbol = SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, fillColor, null)
+    resultGraphicsOverlay.renderer = SimpleRenderer(fillSymbol)
+
+    val pointSymbol = SimpleMarkerSymbol(
+      SimpleMarkerSymbol.Style.CIRCLE,
+      Color.RED,
+      10F
+    )
+    inputGraphicsOverlay.renderer = SimpleRenderer(pointSymbol)
+
+    // add graphics overlays to the map view
+    mapView.graphicsOverlays.apply {
+      add(resultGraphicsOverlay)
+      add(inputGraphicsOverlay)
+    }
+
+    mapView.onTouchListener = object : DefaultMapViewOnTouchListener(
+      applicationContext,
+      mapView
+    ) {
+      override fun onSingleTapConfirmed(e: MotionEvent): Boolean {
+        val screenPoint = android.graphics.Point(
+          e.x.roundToInt(),
+          e.y.roundToInt()
+        )
+        val mapPoint = mMapView.screenToLocation(screenPoint)
+        addGraphicForPoint(mapPoint)
+        calculateViewshedAt(mapPoint)
+        return super.onSingleTapConfirmed(e)
+      }
+    }
+  }
+
+  /**
+   * Adds a graphic at the chosen mapPoint.
+   *
+   * @param point in MapView coordinates.
+   */
+  private fun addGraphicForPoint(point: Point) {
+    // remove existing graphics
+    inputGraphicsOverlay.graphics.clear()
+
+    // add new graphic to the graphics overlay
+    inputGraphicsOverlay.graphics.add(Graphic(point))
+  }
+
+  /**
+   * Uses the given point to create a FeatureCollectionTable which is passed to performGeoprocessing.
+   *
+   * @param point in MapView coordinates.
+   */
+  private fun calculateViewshedAt(point: Point) {
```

Comment:
I know this is probably the same name as in Java...but maybe it's slightly more readable to call it:

calculateViewshedFrom(point) ?

Reply:
I think that reads better